adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/impl/Application.java

```diff
@@ -22,7 +22,9 @@
 /**
  * The Hono MQTT adapter main application class.
  */
-@ComponentScan(basePackages = { "org.eclipse.hono.adapter.mqtt", "org.eclipse.hono.service.metric" })
+@ComponentScan("org.eclipse.hono.adapter.mqtt")
+@ComponentScan("org.eclipse.hono.adapter.mqtt.impl")
```

Comment:
This should already be covered by *base package* `org.eclipse.hono.adapter.mqtt`, shouldn't it?

Reply:
I wasn't able to find any explicit documentation, but running a test confirmed that this does work. So I dropped the annotations for "impl".